registry/handlers/referrers.go

```diff
@@ -59,13 +64,30 @@ func (h *referrersHandler) GetReferrers(w http.ResponseWriter, r *http.Request)
 		return
 	}
 
+	// extract the artifactType filter.
 	var annotations map[string]string
 	var artifactTypeFilter string
 	if artifactTypeFilter = r.URL.Query().Get("artifactType"); artifactTypeFilter != "" {
 		annotations = map[string]string{
 			v1.AnnotationReferrersFiltersApplied: "artifactType",
 		}
 	}
+
+	// extract the page size info. Users define page size by using the n
+	// query parameter. The default page size is 100.
+	pageSize, nParseError := strconv.Atoi(r.URL.Query().Get("n"))
+	if nParseError != nil || pageSize < minPageSize || pageSize > maxPageSize {
+		pageSize = maxPageSize
+	}
+
+	// extract the page number info.
+	pageNumber, pParseError := strconv.Atoi(r.URL.Query().Get("p"))
```

Comment:
We don't need to name each error with a different name.
```suggestion
	pageNumber, err:= strconv.Atoi(r.URL.Query().Get("p"))
```

Reply:
changed accordingly.